Search/Metadata/FieldEvaluator.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Massive\Bundle\SearchBundle\Search\Metadata;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Massive\Bundle\SearchBundle\Search\Metadata\Field\Property;
+use Massive\Bundle\SearchBundle\Search\Metadata\Field\Field;
+use Massive\Bundle\SearchBundle\Search\Metadata\Field\Expression;
+
+/**
+ * Evaluate the value of fields
+ */
+class FieldEvaluator
+{
+    /**
+     * @var ExpressionLanguage
+     */
+    private $expressionLanguage;
+
+    /**
+     * @var PropertyAccess
+     */
+    private $accessor;
+
+    /**
+     * @param Factory $factory
+     * @param ExpressionLanguage $expressionLanguage
+     */
+    public function __construct(ExpressionLanguage $expressionLanguage)
+    {
+        $this->expressionLanguage = $expressionLanguage;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * Evaluate the value from the given object and field
+     *
+     * @param mixed $object
+     * @param mixed $field
+     */
+    public function getValue($object, FieldInterface $field)
+    {
+        try {
+            switch (get_class($field)) {
+                case 'Massive\Bundle\SearchBundle\Search\Metadata\Field\Property':
+                    return $this->getPropertyValue($object, $field);
+                case 'Massive\Bundle\SearchBundle\Search\Metadata\Field\Expression':
+                    return $this->getExpressionValue($object, $field);
+                case 'Massive\Bundle\SearchBundle\Search\Metadata\Field\Field':
+                    return $this->getFieldValue($object, $field);
+            }
```

Comment:
That would push the minimum php version up to 5.5. This would suck, for example my server only has 5.4